core/services/keeper/registry_synchronizer_core.go

```diff
@@ -128,22 +109,22 @@ func (rs *RegistrySynchronizer) Close() error {
 }
 
 func (rs *RegistrySynchronizer) run() {
-	syncTicker := time.NewTicker(rs.interval)
-	logTicker := time.NewTicker(time.Second)
+	syncTicker := utils.NewResettableTimer()
 	defer rs.wgDone.Done()
 	defer syncTicker.Stop()
-	defer logTicker.Stop()
 
 	rs.fullSync()
 
 	for {
 		select {
 		case <-rs.chStop:
 			return
-		case <-syncTicker.C:
+		case <-syncTicker.Ticks():
 			rs.fullSync()
-		case <-logTicker.C:
+			syncTicker.Reset(rs.interval)
+		case <-rs.mbLogs.Notify():
 			rs.processLogs()
+			syncTicker.Reset(rs.interval)
```

Comment:
Do we need to reset the ticker when any single log comes in? This assumes that a full sync doesn't need to occur if we are receiving logs and should occur if `interval` time passes without logs, right?

Reply:
Exactly, yeah. Receive log -> reset the wait time. Probably will save node ops a little money on RPC calls too